From 8867df7b11cb3888a03925d7f3231ca78f8ac14f Mon Sep 17 00:00:00 2001
From: Tristan Cacqueray <tdecacqu@redhat.com>
Date: Tue, 12 Jul 2016 15:32:40 -0400
Subject: [PATCH] Call parameter_function before swift parameter

This change allows dynamic setting of LOG_PATH url
before computing swift hmac signature.

Change-Id: I507905083563007c106715b7b4fa754edf11ac5f
---

diff --git a/zuul/launcher/gearman.py b/zuul/launcher/gearman.py
index 98307ee..10f1025 100644
--- a/zuul/launcher/gearman.py
+++ b/zuul/launcher/gearman.py
@@ -227,6 +227,16 @@
     def updateBuildParams(self, job, item, params):
         """Allow the job to modify and add build parameters"""
 
+        if callable(job.parameter_function):
+            pargs = inspect.getargspec(job.parameter_function)
+            if len(pargs.args) == 2:
+                job.parameter_function(item, params)
+            else:
+                job.parameter_function(item, job, params)
+            self.log.debug("Custom parameter function used for job %s, "
+                           "change: %s, params: %s" % (job, item.change,
+                                                       params))
+
         # NOTE(jhesketh): The params need to stay in a key=value data pair
         # as workers cannot necessarily handle lists.
 
@@ -259,16 +269,6 @@
                 # given  in the form of NAME_PARAMETER=VALUE
                 for key, value in swift_instructions.items():
                     params['_'.join(['SWIFT', name, key])] = value
-
-        if callable(job.parameter_function):
-            pargs = inspect.getargspec(job.parameter_function)
-            if len(pargs.args) == 2:
-                job.parameter_function(item, params)
-            else:
-                job.parameter_function(item, job, params)
-            self.log.debug("Custom parameter function used for job %s, "
-                           "change: %s, params: %s" % (job, item.change,
-                                                       params))
 
     def launch(self, job, item, pipeline, dependent_items=[]):
         uuid = str(uuid4().hex)
